### PR TITLE
reference/best-practices: add Haproxy Best Practice

### PR DESCRIPTION
**Pull request by superlzs0476**
### What is changed, added or deleted? 

add best-practices of HAProxy to docs-cn.

### What is the related PR or file link(s)? 

#1796 (the original PR)

### CODE REVIEW

**Comment by superlzs0476** [comment]
I have created a new one of PR  for "Add HAProxy Best Practics", because of before the PR  has too many errors about CI  inspector. 
@juliezhang1112 @lilin90  PTAL

File: v3.0/reference/best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+## 部署 HAProxy
+
+HAProxy 配置 Database 负载均衡场景操作简单，以下部署操作具有普遍性，不具有特殊性，建议根据实际场景，个性化配置相关的[配置文件](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html)。
+
+> **注意：**
+>
+> 官方目前建议使用 HAProxy 稳定版本 2.0，其特性参见 [2.0 特性](https://www.haproxy.com/blog/haproxy-2-0-and-beyond/)。
+
+![HAProxy](/media/Haproxy.jpg)
```
Comment (yikeke) [inline v3.0/reference/best-practices/haproxy-best-practices.md]:
```suggestion
![HAProxy](/media/haproxy.jpg)
```

File: v3.0/reference/best-practices/haproxy-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+|-dM|[\<byte>] poisons memory with \<byte> (defaults to 0x50)|执行分配内存|
+|-V|enters verbose mode (disables quiet mode)|启动过程显示配置和轮询信息
+|-D|goes daemon|开启守护进程模式||
+|-C|changes to \<dir> before loading files.||
+|-W|master-worker mode.|主从模式|
+|-q|quiet mode: don't display messages|静默模式，不输出信息|
+|-c|check mode: only checks config files and exits|检查配置信息文件|
+|-n|sets the maximum total # of connections (2000)|最大总连接数|
+|-m|limits the usable amount of memory (in MB)|最大使用内存|
+|-N|sets the default, per-proxy maximum # of connections (2000)|单点最大连接数|
```
Comment (juliezhang1112) [inline v3.0/reference/best-practices/haproxy-best-practices.md]:
```suggestion
|-N|sets the default, per-proxy maximum # of connections (2000)|设置单点最大连接数，默认为2000 |
```

File: v3.0/reference/best-practices/haproxy-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+|-L|sets local peer name (default to hostname)|本地实例对等名称|
+|-p|writes pids of all children to this file|HAProxy 进程 pid 信息写入 file |
```
Comment (juliezhang1112) [inline v3.0/reference/best-practices/haproxy-best-practices.md]:
```suggestion
|-p|writes pids of all children to this file|将HAProxy 所有子进程的 PID 信息写入该文件 |
```

**Comment by yikeke** [comment]
Please help check whether the changes in [this commit](https://github.com/pingcap/docs-cn/pull/1889/commits/e9b8cde64646cf224103668cdbbfad2a6c17b82c) are correct, thanks~ @superlzs0476

**Comment by juliezhang1112** [comment]
PTAL @cocoyi17 
Thanks！

**Pull request merged**